consensus/src/dag/rb_handler.rs

```diff
@@ -67,22 +72,24 @@ impl NodeBroadcastHandler {
         self.storage.delete_votes(to_delete)
     }
 
-    fn validate(&self, node: &Node) -> anyhow::Result<()> {
+    fn validate(&self, node: Node) -> anyhow::Result<Node> {
         let current_round = node.metadata().round();
 
         // round 0 is a special case and does not require any parents
         if current_round == 0 {
-            return Ok(());
+            bail!(NodeBroadcastHandleError::InvalidRound);
```

Comment:
nit: we should start to carry more debug information into the errors, like exact round, missing parents id, why it's invalid etc

Reply:
Fix this now. I will do a separate pass for improving the errors overall and start adding metric counters.